scripts/language_model/word_language_model.py

```diff
@@ -362,67 +373,105 @@ def train():
             target_list = gluon.utils.split_and_load(target, context, batch_axis=1, even_split=True)
             hiddens = detach(hiddens)
             Ls = []
-            L = 0
             with autograd.record():
                 for j, (X, y, h) in enumerate(zip(data_list, target_list, hiddens)):
-                    output, h, encoder_hs, dropped_encoder_hs = model.forward(X, h)
+                    output, h, encoder_hs, dropped_encoder_hs = model(X, h)
                     l = joint_loss(output, y, encoder_hs, dropped_encoder_hs)
-                    L = L + l.as_in_context(context[0]) / X.size
-                    Ls.append(l/X.size)
+                    Ls.append(l.as_in_context(context[0]) / (len(context) * X.size))
                     hiddens[j] = h
-            L.backward()
-            grads = [p.grad(d.context) for p in parameters for d in data_list]
-            gluon.utils.clip_global_norm(grads, args.clip)
+            for L in Ls:
+                L.backward()
 
-            trainer.step(1)
+            nlp.model.utils.multi_gpu_clip_global_norm(trainer, parameters.values(), args.clip)
 
-            total_L += sum([mx.nd.sum(L).asscalar() for L in Ls]) / len(context)
+            if args.ntasgd and ntasgd:
+                if param_dict_avg is None:
+                    param_dict_avg = {k.split(model._prefix)[1]: v.data(context[0]).copy()
```

Comment:
I think there is no need to remove the model prefix. Otherwise, you cannot use `load_parameters` directly.

Reply:
We should, I have verified.